resolwe/flow/serializers/base.py

```diff
@@ -56,3 +59,10 @@ def __init__(self, instance=None, data=empty, **kwargs):
     def request(self):
         """Extract request object from serializer context."""
         return self.context.get('request', None)
+
+    def save(self):
+        """Override save method to catch handled errors and repackage them as 400 errors."""
+        try:
+            super(ResolweBaseSerializer, self).save()
+        except DatabaseError as database_error:
```

Comment:
Not the best idea.

A better solution would be to raise a specific resolwe error [here](https://github.com/genialis/resolwe/blob/7b8ee3e4de21affa7cf18d73e9f0acb5556c06ec/resolwe/flow/models/fields.py#L178) and catch it here. That would entail an `excpetions` package with that error defined.

Reply:
Actually a really bad idea :).

Create a custom error and put it in `flow/exceptions.py`.